rpc/core/tx.go

```diff
@@ -205,7 +205,8 @@ func TxSearch(query string, prove bool, page, perPage int) (*ctypes.ResultTxSear
 
 	apiResults := make([]*ctypes.ResultTx, cmn.MinInt(perPage, totalCount-skipCount))
 	var proof types.TxProof
-	for i := 0; i < len(apiResults); i++ {
+	// if there's no tx in the results array, we don't need to loop through the apiResults array
+	for i := 0; i < len(apiResults) && len(results) > 0; i++ {
```

Comment:
Is this really needed? The test case above seems to cover a different use-case.

if `totalCount` is 0, then the `len(apiResults)` will also be 0

Reply:
Not necessarily @melekes, I had this issue myself when I had to query the page 1 returning one item per page.

The app crashes returning `index out of range` because of this line:
`r := results[skipCount+i]`

if results is empty, I get that error.

You can reproduce it by removing the `&& len(results)` and running the test I created.

Reply:
oh okay

Reply:
all good ?

Reply:
yes